source/parser/unary.civet

```diff
@@ -53,19 +53,24 @@ function processUnaryExpression(pre, exp, post)
         }
 
   // Await ops
-  const l = pre.length
-  if (l) {
-    const last = pre[l - 1]
-    if (last.type is "Await" and last.op) {
-      if (exp.type !== "ParenthesizedExpression") {
-        exp = ["(", exp, ")"]
-      }
-      exp = {
-        type: "CallExpression",
-        children: [" Promise", last.op, exp]
-      }
-    }
-  }
+  while l := pre.length
+    last := pre[l - 1]
+    if last.type is "Await"
+      if last.op
+        if exp.type !== "ParenthesizedExpression"
+          exp = ["(", exp, ")"]
+
+        exp =
+          type: "CallExpression",
+          children: [...last.children, "Promise", last.op, exp]
+        pre = pre.slice(0, -1)
```

Comment:
`pre.pop()` (in place) would probably be more efficient. Or are we not allowed to modify here?

(also 5 lines down)

Reply:
We shouldn't modify here since this is called from the parser sometimes. There's usually only zero or one await ops per expression as well.

Reply:
Perhaps we should start adding `readonly` type annotations for functions we call from the compiler. I believe this can implement checks for this sort of thing.